### PR TITLE
feat: [CDS-44917]: Freeze Window - Visual Section Tests

### PR DESCRIPTION
**Pull request by ishantmehtaharness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by sonaligoyal-harness — APPROVED** [review]

File: src/modules/72-freeze-windows/components/FreezeWindowStudio/__tests__/FreezeWindowStudioOverviewSection.test.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    await userEvent.click(getByText('continue'))
+    await waitFor(() => {
+      expect(getByText('common.validation.nameIsRequired')).toBeInTheDocument()
+    })
+    expect(onNext).not.toBeCalled()
+    expect(container).toMatchSnapshot('Overview Section Snapshot')
+    await fillAtForm([{ container, fieldId: 'name', type: InputTypes.TEXTFIELD, value: 'Weekend Freeze' }])
+    expect(updateFreeze).toHaveBeenCalledWith({ name: 'Weekend Freeze', identifier: 'Weekend_Freeze' })
+    await userEvent.click(getByText('continue'))
+    // expect(onNext).toHaveBeenCalled()
```
Comment (yc147) [inline src/modules/72-freeze-windows/components/FreezeWindowStudio/__tests__/FreezeWindowStudioOverviewSection.test.tsx:63]: What's the assertion here?

**Review by vkbansal-harness — APPROVED** [review]

File: src/modules/72-freeze-windows/components/FreezeWindowStudio/__tests__/FreezeWindowStudioOverviewSection.test.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    useParams.mockImplementation(() => {
+      return { projectIdentifier, orgIdentifier, accountId, module: 'cd', windowIdentifier: '-1' }
+    })
+  })
+
+  test('it should render Overview section in create mode', async () => {
+    const updateFreeze = jest.fn()
+    const onNext = jest.fn()
+    const { getByText, container } = render(
+      <TestWrapper>
```
Comment (yc147) [inline src/modules/72-freeze-windows/components/FreezeWindowStudio/__tests__/FreezeWindowStudioOverviewSection.test.tsx:41]: Any specific reason for mocking `useParams` ? We could have just `path` and `pathParams` in `TestWrapper`.

File: src/modules/72-freeze-windows/components/FreezeWindowStudio/__tests__/FreezeWindowStudioOverviewSection.test.tsx (hunk 1 above)
Comment (ishantmehtaharness) [reply src/modules/72-freeze-windows/components/FreezeWindowStudio/__tests__/FreezeWindowStudioOverviewSection.test.tsx:63]: This is commented for now. I am implementing this assertion in next PR

File: src/modules/72-freeze-windows/components/FreezeWindowStudio/__tests__/FreezeWindowStudioOverviewSection.test.tsx (hunk 2 above)
Comment (ishantmehtaharness) [reply src/modules/72-freeze-windows/components/FreezeWindowStudio/__tests__/FreezeWindowStudioOverviewSection.test.tsx:41]: no specific reason.
I can explore pathParams in next PR

File: src/modules/72-freeze-windows/components/FreezeWindowStudio/__tests__/FreezeWindowStudioOverviewSection.test.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      </TestWrapper>
+    )
+
+    expect(getByText('freezeWindows.freezeStudio.freezeOverview')).toBeInTheDocument()
+    await userEvent.click(getByText('continue'))
+    await waitFor(() => {
+      expect(getByText('common.validation.nameIsRequired')).toBeInTheDocument()
+    })
+    expect(onNext).not.toBeCalled()
+    expect(container).toMatchSnapshot('Overview Section Snapshot')
```
Comment (chetanvekariya-harness) [inline src/modules/72-freeze-windows/components/FreezeWindowStudio/__tests__/FreezeWindowStudioOverviewSection.test.tsx:47]: Please avoid writing snapshot tests as much as possible. It creates lot of chaos when smallest of styling is changed. Adding two to three assertions would be much better than writing snapshot assertion.

File: src/modules/72-freeze-windows/components/FreezeWindowStudio/__tests__/FreezewindowStudioBody.test.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    const allTabs = container.getElementsByClassName('bp3-tab')
+    expect(allTabs.length).toBe(3)
+    expect(getByText('overview')).toBeInTheDocument()
+    expect(getByText('freezeWindows.freezeStudio.freezeConfiguration')).toBeInTheDocument()
+    expect(getByText('common.schedule')).toBeInTheDocument()
+
+    // Freeze Overview Section should be defined
+    expect(getByText('freezeWindows.freezeStudio.freezeOverview')).toBeInTheDocument()
+    expect(container).toMatchSnapshot('Overview Tab snapshot')
```
Comment (chetanvekariya-harness) [inline src/modules/72-freeze-windows/components/FreezeWindowStudio/__tests__/FreezewindowStudioBody.test.tsx:46]: Same as other comment that is made regarding snapshot testing above.

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/72-freeze-windows/components/FreezeWindowStudio/__tests__/FreezeWindowStudioOverviewSection.test.tsx (hunk 2 above)
Comment (ishantmehtaharness) [reply src/modules/72-freeze-windows/components/FreezeWindowStudio/__tests__/FreezeWindowStudioOverviewSection.test.tsx]: This is done

File: src/modules/72-freeze-windows/components/FreezeWindowStudio/__tests__/FreezeWindowStudioOverviewSection.test.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    await fillAtForm([{ container, fieldId: 'name', type: InputTypes.TEXTFIELD, value: 'Weekend Freeze' }])
+    expect(updateFreeze).toHaveBeenCalledWith({ name: 'Weekend Freeze', identifier: 'Weekend_Freeze' })
+    await userEvent.click(getByText('continue'))
```
[the remark this answers is not in the document]
Comment (ishantmehtaharness) [reply src/modules/72-freeze-windows/components/FreezeWindowStudio/__tests__/FreezeWindowStudioOverviewSection.test.tsx:50]: Yes, sure. adding in subsequent PR's

File: src/modules/72-freeze-windows/components/FreezeWindowStudio/__tests__/FreezeWindowStudioOverviewSection.test.tsx (hunk 3 above)
Comment (ishantmehtaharness) [reply src/modules/72-freeze-windows/components/FreezeWindowStudio/__tests__/FreezeWindowStudioOverviewSection.test.tsx:47]: @chetanvekariya-harness I completely agree with you on this one
Although, I have added snapshots for all components exactly once.
This is done in order to have the dom tree covered.

Also, when small things changes, the snapshots really help.
I can tell you by my experience, when I was upgrading Formik. Snapshots were really helpful to me, helped me find 5 critical bugs, which I would have missed, if we didnt had snapshots

**Review by vkbansal-harness — APPROVED** [review]

**Pull request merged**